pkg/cli/mirror/cincinnati_graph_image.go

```diff
@@ -184,3 +192,67 @@ func downloadGraphData(ctx context.Context, dir string) error {
 	_, err = io.Copy(out, resp.Body)
 	return err
 }
+
+func copySignatureForUpdateGraph(srcSigDir string, dstGraphDataDir string) error {
+
+	//Go through the files in srcSignaturePath and parse the json files.
+	files, err := os.ReadDir(srcSigDir)
+	if err != nil {
+		return err
+	}
+	for _, file := range files {
+		if file.IsDir() {
+			continue
+		}
+		//Read the file
+		signatureRawBytes, err := os.ReadFile(filepath.Join(srcSigDir, file.Name()))
+		if err != nil {
+			return err
+		}
+		//Parse the file
+		cmObj, err := util.ReadConfigMap(signatureRawBytes)
+		if err != nil {
+			continue
+		}
+		if cmObj == nil {
+			continue
+		}
+		//Copy the signature to graph data directory
+		err = copySignatureToGraphDataDir(dstGraphDataDir, cmObj)
+		if err != nil {
+			return fmt.Errorf("error copying signatures to Cincinnati graph data directory: %v", err)
+		}
+	}
+	return nil
+}
+
+// copySignatureToGraphDataDir() will copy the signatures to cincinnati graph data directory
+// with signatures/{algorithm}/{digest}/signature-{number} schema.
+func copySignatureToGraphDataDir(graphDataSignatureDir string, cmObj *corev1.ConfigMap) error {
+
+	sigDirPath := filepath.Join(graphDataSignatureDir, "signatures")
+
+	//iterate through the cmObj.BinaryData map
+	for key, value := range cmObj.BinaryData {
+		//example key = sha256-73946971c03b43a0dc6f7b0946b26a177c2f3c9d37105441315b4e3359373a55-1
+		v := strings.Split(key, "-")
```

Comment:
@LalatenduMohanty - would the spliit of ("-") always be  the case ?
Could you maybe add a len check  of v ? to avoid a panic

Reply:
Fixed as part of https://github.com/openshift/oc-mirror/commit/0c9a949f60ca6d711a9dfa3be4c82823aa51d9b8